distsql/select_result.go

```diff
@@ -456,6 +454,9 @@ func (r *selectResult) Close() error {
 	if respSize > 0 {
 		r.memConsume(-respSize)
 	}
+	if r.stats != nil {
+		defer r.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.RegisterStats(r.rootPlanID, r.stats)
```

Comment:
why use `defer` here?

Reply:
Use defer to ensure that the `Close()`(make sure all goroutines have been exit in some parallel executer) is finished before registering stats.